programs/drift/src/math/auction.rs

```diff
@@ -76,37 +78,118 @@ pub fn calculate_auction_prices(
+        OrderType::Market | OrderType::TriggerMarket => {
+            calculate_auction_price_for_fixed_auction(order, slot, tick_size)
+        }
+        OrderType::Oracle => calculate_auction_price_for_oracle_offset_auction(
+            order,
+            slot,
+            tick_size,
+            valid_oracle_price,
+        ),
+        _ => unreachable!(),
+    }
+}
+
+fn calculate_auction_price_for_fixed_auction(
+    order: &Order,
+    slot: u64,
+    tick_size: u64,
+) -> DriftResult<u64> {
     let slots_elapsed = slot.safe_sub(order.slot)?;
 
     let delta_numerator = min(slots_elapsed, order.auction_duration.cast()?);
     let delta_denominator = order.auction_duration;
 
+    let auction_start_price = order.auction_start_price.cast::<u64>()?;
+    let auction_end_price = order.auction_end_price.cast::<u64>()?;
+
     if delta_denominator == 0 {
-        return Ok(order.auction_end_price);
+        return standardize_price(auction_end_price, tick_size, order.direction);
     }
 
     let price_delta = match order.direction {
-        PositionDirection::Long => order
-            .auction_end_price
-            .safe_sub(order.auction_start_price)?
+        PositionDirection::Long => auction_end_price
+            .safe_sub(auction_start_price)?
             .safe_mul(delta_numerator.cast()?)?
             .safe_div(delta_denominator.cast()?)?,
-        PositionDirection::Short => order
-            .auction_start_price
-            .safe_sub(order.auction_end_price)?
+        PositionDirection::Short => auction_start_price
+            .safe_sub(auction_end_price)?
             .safe_mul(delta_numerator.cast()?)?
             .safe_div(delta_denominator.cast()?)?,
     };
 
     let price = match order.direction {
-        PositionDirection::Long => order.auction_start_price.safe_add(price_delta)?,
-        PositionDirection::Short => order.auction_start_price.safe_sub(price_delta)?,
+        PositionDirection::Long => auction_start_price.safe_add(price_delta)?,
+        PositionDirection::Short => auction_start_price.safe_sub(price_delta)?,
     };
 
     standardize_price(price, tick_size, order.direction)
 }
 
+fn calculate_auction_price_for_oracle_offset_auction(
+    order: &Order,
+    slot: u64,
+    tick_size: u64,
+    valid_oracle_price: Option<i64>,
+) -> DriftResult<u64> {
+    let oracle_price = valid_oracle_price.ok_or_else(|| {
+        msg!("Could not find oracle too calculate oracle offset auction price");
+        ErrorCode::OracleNotFound
+    })?;
+
+    let slots_elapsed = slot.safe_sub(order.slot)?;
+
+    let delta_numerator = min(slots_elapsed, order.auction_duration.cast()?);
+    let delta_denominator = order.auction_duration;
+
+    let auction_start_price_offset = order.auction_start_price;
+    let auction_end_price_offset = order.auction_end_price;
+
+    if delta_denominator == 0 {
+        let price = oracle_price.safe_add(auction_end_price_offset)?;
+
+        if price <= 0 {
```

Comment:
would it make sense to let price = 0 in this case and not throw an error?

Reply:
im not sure we should allow limit price to equal zero, was also thinking we might want to add an assert in get_limit_price that it's not 0 @0xbigz ?

Reply:
dont understand the reason to support price=0